tests: drivers: build_all: led_strip: Add i2c-devices build test

soburi opened the pull request:
Add build tests for following devices.

- apa,apa102
- greeled,lpd8806
- worldsemi,ws2812-spi

soburi committed:
tests: drivers: build_all: led_strip: Add i2c-devices build test

Add build tests for following devices.

- apa,apa102
- greeled,lpd8806
- worldsemi,ws2812-spi

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi commented on tests/drivers/build_all/led_strip/app.overlay:
```diff
@@ -35,7 +35,9 @@
-			cs-gpios = <&test_gpio 0 0>;
+			cs-gpios = <&test_gpio 0 0>,
+				   <&test_gpio 0 0>,
+				   <&test_gpio 0 0>;
```
The reg value was wrong, so we need four.
The device address corresponds to the array element.
All four of these are SPI devices., and a corresponding cs definition is required for each.

The build_all test is simply a test to check compilation and does not consider behavior at runtime.

simonguinot replied:
That's why I was surprised to only see 3 CS lines :)

soburi committed:
drivers: led_strip: tlc5971: Enable when the dt-node exists

The TLC5971_STRIP config enables if the "ti,tlc5971" node exists.
Also, removing the setting that explicitly enables in the
`tests/drivers/build_all/led_strip` test.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi committed:
tests: drivers: build_all: led_strip: Add RaspberryPi Pico PIO driver

Add build tests for "worldsemi,ws2812_rpi_pico_pio" driver

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>
